framework/src/media/utils/MediaUtils.cpp

```diff
@@ -43,6 +150,393 @@ static const std::string MP4_MIME_TYPE = "audio/mp4";
+static int findTsPacket(const uint8_t *buffer, size_t size, bool sync)
+{
+	if (size < TS_PACKET_SIZE) {
+		return UTILS_ERROR_WANT_MORE_DATA;
+	}
+	size_t syncOffset = 0;
+	if (sync || (buffer[0] != TS_SYNC_BYTE)) {
+		for (syncOffset = 0; syncOffset < TS_PACKET_SIZE; syncOffset++) {
+			if (buffer[syncOffset] != TS_SYNC_BYTE) {
+				continue;
+			}
+			if (size < syncOffset + TS_SYNC_COUNT * TS_PACKET_SIZE) {
+				return UTILS_ERROR_WANT_MORE_DATA; // not enough for sync
+			}
+			int count;
+			for (count = 1; count < TS_SYNC_COUNT; count++) {
+				if (buffer[syncOffset + count * TS_PACKET_SIZE] != TS_SYNC_BYTE) {
+					break;
+				}
+			}
+			if (count == TS_SYNC_COUNT) {
+				break; // sync succeed
+			}
+		}
+		if (syncOffset == TS_PACKET_SIZE) {
+			return UTILS_ERROR_NOT_FOUND; // sync byte not found
+		}
+	}
+	return (int)syncOffset;
+}
+
+// Abstract payload from the given TS packet
+static const uint8_t *getPayloadData(const uint8_t *packet, uint8_t *payloadDataLen)
+{
+	uint8_t lenPayload = TS_PACKET_SIZE - TS_HEAD_BYTES;
+	const uint8_t *payload = packet + TS_HEAD_BYTES;
+	if (TS_TRANSPORT_ERROR_INDICATOR(packet)) {
+		meddbg("Transport Error\n");
+		return nullptr;
+	}
+	if (TS_ADAPTION_FIELD_CONTROL(packet) == 0) { // 0 - reserved
+		medwdbg("adaptationFieldControl 0 is reserved\n");
+		return nullptr;
+	}
+	if (TS_ADAPTION_FIELD_CONTROL(packet) == 2) { // 2 - 183bytes adaptation field only
+		return nullptr; // no playload
+	}
+	if (TS_ADAPTION_FIELD_CONTROL(packet) == 3) { // 3 - 1~182bytes adaptation field + payload
+		lenPayload = TS_PACKET_SIZE - TS_HEAD_BYTES - (TS_ADAPT_FIELD_LENGTH_BYTES + TS_ADAPTION_FIELD_LENGTH(packet));
+		payload = packet + (TS_PACKET_SIZE - lenPayload);
+	} // else 1 - 184bytes playload
+	*payloadDataLen = lenPayload;
+	return payload;
+}
+
+// Unpack TS packet to abstract PSI section
+static int PSI_Unpack(const uint8_t *packet, const uint8_t **section, uint16_t *sectlen)
+{
+	uint8_t  lenPayload = 0;
+	const uint8_t *ptrPayload = getPayloadData(packet, &lenPayload);
+	if (!ptrPayload) {
+		// no payload
+		*section = NULL;
+		return UTILS_ERROR_NONE;
+	}
+	if (TS_PAYLOAD_UNIT_START_INDICATOR(packet)) {
+		// new section start, and the 1st byte in payload is the pointer field.
+		uint8_t u8PointerField = ptrPayload[0];
+		if (u8PointerField != 0) {
+			// prev section tail and next section head in this packet!
+			return UTILS_ERROR_NOT_SUPPORTED;
+		}
+		// handle new section
+		const uint8_t *ptrSection = ptrPayload + 1 + u8PointerField;
+		uint16_t lenSection = (SECTION_HEAD_BYTES + SECTION_LENGTH(ptrSection));
+		if (lenSection > (lenPayload - 1 - u8PointerField)) {
+			// portion of section in this packet
+			return UTILS_ERROR_NOT_SUPPORTED;
+		}
+		// TODO: check CRC32
+		// section complete (whole section contained in one TS packet)
+		*section = ptrSection;
+		*sectlen = lenSection;
+		return UTILS_ERROR_NONE;
+	} else {
```

Comment:
always return for if case, so this else seems unnecessary here

Reply:
Fixed, thanks